staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -489,10 +495,14 @@ func (cnc *CloudNodeController) syncNode(ctx context.Context, nodeName string) e
 			return err
 		}
 
+		removeCloudProviderTaintDelay.Observe(time.Since(newNode.ObjectMeta.CreationTimestamp.Time).Seconds())
+
 		// After adding, call UpdateNodeAddress to set the CloudProvider provided IPAddresses
 		// So that users do not see any significant delay in IP addresses being filled into the node
 		cnc.updateNodeAddress(ctx, newNode, instanceMetadata)
 
+		initializeNodeAddressDelay.Observe(time.Since(newNode.ObjectMeta.CreationTimestamp.Time).Seconds())
```

Comment:
we only want to add this metric if we actually change the address, that is why it has to go in line 405, after
```
	if _, _, err := nodeutil.PatchNodeStatus(cnc.kubeClient.CoreV1(), types.NodeName(node.Name), node, newNode); err != nil {
		klog.Errorf("Error patching node with cloud ip addresses = [%v]", err)
	}
```

Reply:
There's no way to tell within that function if we're doing the initial population of the node's addresses or a periodic reconciliation. I could add a return value (bool) that indicates whether the addresses were actually modified, and only emit the metric when true?

Reply:
Added in the latest revision, lmk your thoughts 👍